### PR TITLE
`GeodatabaseTransactions` fixes

### PR DESCRIPTION
**Pull request by williambohrmann3**
# Description

Fixes:
* Handles exception when adding geometry outside of geodatabase extent.
* Ensures geometry editor is stopped and property changed event is unsubscribed.
* Checks for local edits before attempting to sync.

Code style:
* Follows established `Initialize()` pattern.
* Uses `var` during object instantiation.
* Limits `GeometryEditor.PropertyChanged` event subscriptions.

UIX:
* Symbolizes geodatabase extent with red rectangle.
* Reports exceptions to the control panel.
* Handles case when there are no edits to commit.

## Type of change

- Bug fix
- Other enhancement

## Platforms tested on

- [x] WPF .NET 8
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [ ] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

**Comment by williambohrmann3** [comment]
cc: @cgutierrez-gis

File: src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs [hunk 1]
```diff
@@ -41,22 +43,53 @@ public partial class GeodatabaseTransactions : ContentPage
...
+
+                // Create a graphics overlay for the extent graphic and apply a renderer.
+                var extentOverlay = new GraphicsOverlay
+                {
+                    Graphics = { extentGraphic },
+                    Renderer = new SimpleRenderer(lineSymbol)
+                };
+
+                // Add graphics overlay to the map view.
+                MyMapView.GraphicsOverlays.Add(extentOverlay);
```
Comment (jnery) [inline src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:88]: I think adding the GraphicsOverlay with SimpleRenderer do not need to wait for SR changed and can be done upfront.

What may need to wait is adding a graphic with the extent geometry. If this happens outside try-catch, is there guarantee/check `_extent`  is not null or empty?

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs [hunk 2]
```diff
@@ -44,10 +48,19 @@ public partial class GeodatabaseTransactions
         // Store a reference to the table to be edited.
         private GeodatabaseFeatureTable _editTable;
 
+        // Flag indicating if an edit exists which is not committed and was created during a transaction.
+        // Not to be confused with HasLocalEdits() which is for local edits not synchronized with the service.
+        private bool _hasUncommittedEdits;
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:53]: I don't think we need this. `Geodatabase.HasLocalEdits()` or `GeodatabaseFeatureTable.HasLocalEdits()` can be used instead.

**Review by jnery — CHANGES_REQUESTED** [review]

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs (hunk 2 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:53]: This handles the case where user has unsynced edits (HasLocalEdits() == true), so initially they created a feature. Then, they tap add feature button, but before they tap the map, they tap stop editing button. The prompt makes it seems like there are changes to commit locally, but there isn't anything from that editing session. Those changes were already locally committed in a previous transaction. Let me know if we want to address this differently.

use-case

File: src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs (hunk 1 above)
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:88]: Good eye! Moving this out of the event handler. We can mark `_extent` as `readonly`.

File: src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs [hunk 3]
```diff
@@ -253,66 +296,82 @@ private void AddNewFeature(object sender, EventArgs args)
...
-                        // If there is, commit the transaction to store the edits (this will also end the transaction).
-                        _localGeodatabase.CommitTransaction();
-                        MessageTextBlock.Text = "Edits were committed to the local geodatabase.";
-                    }
+                    // Commit the transaction to store the edits (this will also end the transaction).
+                    _localGeodatabase.CommitTransaction();
+                    MessageTextBlock.Text = "Edits were committed to the local geodatabase.";
                 }
-                else if (choice == "Rollback")
+                else if (choice == "Rollback") // TODO: remove nested if on other implementations
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:365]: Should update this or remove comment.

Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs]: sorry about that, good eye

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs [hunk 4]
```diff
@@ -245,64 +284,78 @@ private void AddNewFeature(object sender, RoutedEventArgs args)
...
 
         private void StopEditTransaction(object sender, RoutedEventArgs e)
         {
+            // Ensure the geometry editor is stopped and property changed event is unsubscribed since user is leaving editing mode.
+            // Handles case where user stops transaction while geometry editor is active.
+            MyMapView.GeometryEditor.PropertyChanged -= GeometryEditor_PropertyChanged;
+            MyMapView.GeometryEditor.Stop();
+
+            // Handle the case where there are no edits to commit or rollback.
+            if (!_hasUncommittedEdits)
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:330]: I think this should be if `(!_localGeodatabase.HasLocalEdits())`

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs [hunk 5]
```diff
@@ -245,64 +284,78 @@ private void AddNewFeature(object sender, RoutedEventArgs args)
...
+            {
+                MessageTextBlock.Text = "No edits to commit or rollback.";
+
+                // Raise the transaction status changed event to update the UI and reset the flag.
+                _localGeodatabase.RollbackTransaction();
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:335]: We just said no edits to commit/rollback, so let's remove this too

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs (hunk 2 above)
Comment (jnery) [reply src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs:53]: I still think we can do without this property based on how it is used.
May be you want to permit `geometryEditor.Stop()` but do not want g`eodatabase.CommitTransaction()` or `RollbackTransaction()`... in those cases, former can use geometryEditor.IsStarted and latter can use `geodatabase.HasLocalEdits()`

**Review by jnery — CHANGES_REQUESTED** [review]
Sorry I missed the follow-up

File: src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs (hunk 2 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs]: Per our discussion, got rid of this added conditional.

**Review by jnery — APPROVED** [review]

**Pull request merged**